File: cases/taint/go/map-iteration-negative/map_iteration_negative.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: map-iteration-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: map-iteration-sink

func run() {
	carrier := map[string]string{"payload": "clean"}
	other := map[string]string{"payload": "clean"}
	carrier["payload"] = dfb_source() // DFB-WITNESS: map-iteration-store
	_ = carrier
	for _, value := range other {
		dfb_sink(value)
	}
}

File: cases/taint/java/map-iteration-negative/MapIterationNegative.java
package dataflowbench.taint;

import java.util.HashMap;
import java.util.Map;

final class MapIterationNegative {
    static int dfb_source() { // DFB-SOURCE: map-iteration-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: map-iteration-sink

    static void run() {
        Map<String, Integer> tainted = new HashMap<>();
        tainted.put("entry", dfb_source()); // DFB-WITNESS: map-iteration-store
        Map<String, Integer> clean = new HashMap<>();
        clean.put("entry", 0);
        for (Map.Entry<String, Integer> entry : clean.entrySet()) { // DFB-WITNESS: map-iteration-entries
            dfb_sink(entry.getValue());
        }
    }
}
